### PR TITLE
Migrate Mailchimp Source Connector to Low Code

### PR DESCRIPTION
**Pull request by topefolorunso**
## What
Migrating Source Mailchimp to Low-Code CDK resolves #28864

## How
Created the connector using the connector buider UI, then imported the manifest.yaml to the low code cdk.

## Recommended reading order
1. `manifest.yaml`

## 🚨 User Impact 🚨
*Are there any breaking changes? What is the end result perceived by the user?*

*For connector PRs, use this section to explain which type of semantic versioning bump occurs as a result of the changes. Refer to our [Semantic Versioning for Connectors](https://docs.airbyte.com/contributing-to-airbyte/#semantic-versioning-for-connectors) guidelines for more information. **Breaking changes to connectors must be documented by an Airbyte engineer (PR author, or reviewer for community PRs) by using the [Breaking Change Release Playbook](https://docs.google.com/document/d/1VYQggHbL_PN0dDDu7rCyzBLGRtX-R3cpwXaY8QxEgzw/edit).***

*If there are breaking changes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.*

## Pre-merge Actions
*Expand the relevant checklist and delete the others.*

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- Unit & integration tests added

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- Create a non-forked branch based on this PR and test the below items on it
- Build is successful
- If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).

</details>

### CODE REVIEW

File: airbyte-integrations/connectors/source-mailchimp/source_mailchimp/manifest.yaml [hunk 1]
```diff
@@ -0,0 +1,435 @@
...
+        path: lists
+        http_method: GET
+        request_parameters:
+          sort_dir: ASC
+          sort_field: date_created
+        request_headers: {}
+        authenticator:
+          type: BasicHttpAuthenticator
+          password: '{{ config[''password''] }}'
+          username: '{{ config[''username''] }}'
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-mailchimp/source_mailchimp/manifest.yaml:28]: Connector has two type of auth and the implementation is only using one. Please add the other.

File: airbyte-integrations/connectors/source-mailchimp/source_mailchimp/manifest.yaml [hunk 2]
```diff
@@ -0,0 +1,435 @@
...
+        inject_into: request_parameter
+  - type: DeclarativeStream
+    name: automations
+    primary_key:
+      - id
+    retriever:
+      type: SimpleRetriever
+      requester:
+        type: HttpRequester
+        url_base: https://{{ config.password.split('-')[-1] }}.api.mailchimp.com/3.0/
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-mailchimp/source_mailchimp/manifest.yaml:124]: This is not correct. Please check previous implementation.

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
Left some comments. There are some work to be made to this connector migration.

File: airbyte-integrations/connectors/source-mailchimp/source_mailchimp/manifest.yaml [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+    request_parameters:
+      sort_dir: ASC
+      sort_field: '{{ parameters.cursor_field }}'
+    # authenticator:
+    #   class_name: source_mailchimp.components.MailchimpAuthenticator
+      # basic_auth: "#/definitions/basic_authenticator"
+      # oauth: "#/definitions/oauth_authenticator"
+    request_body_json: {}
+    basic_auth: "#/definitions/basic_authenticator"
+    oauth: "#/definitions/oauth_authenticator"
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-mailchimp/source_mailchimp/manifest.yaml:38]: The implementation is not correct.
Please check [this](https://github.com/airbytehq/airbyte/blob/54d40a9c3cbe79e21df5ec0b636f09909bf003ce/airbyte-integrations/connectors/source-retently/source_retently/manifest.yaml#L26-L33)

**Review by marcosmarxm — CHANGES_REQUESTED** [review]

**Comment by marcosmarxm** [comment]
@topefolorunso did you have time to check my comment? Are you going to continue this contribution?

**Comment by topefolorunso** [comment]
> @topefolorunso did you have time to check my comment? Are you going to continue this contribution?

Yes, I am working on it. I am running into errors when I use both custom requester to handle the url_base and the custom authenticator to handle the auth methods

I will share more details soon

**Comment by topefolorunso** [comment]
@marcosmarxm I have pushed my latest changes. Please review. 

The error message I am getting:

` "message": "Error creating custom component source_mailchimp.components.MailchimpRequester. Subcomponent creation has not been implemented for 'DeclarativeAuthenticator'`

File: airbyte-integrations/connectors/source-mailchimp/source_mailchimp/manifest.yaml [hunk 4]
```diff
@@ -0,0 +1,253 @@
...
+      type: DpathExtractor
+      field_path:
+        - '{{ parameters.field_path }}'
+  basic_authenticator:
+      type: BasicHttpAuthenticator
+      password: '{{ config.credentials.apikey }}'
+      username: 'anystring'
+  oauth_authenticator:
+      type: BearerAuthenticator
+      api_token: '{{ config.credentials.access_token }}'
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-mailchimp/source_mailchimp/manifest.yaml:22]: This is not implemented correct.

**Review by marcosmarxm — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connectors/source-mailchimp/source_mailchimp/manifest.yaml (hunk 4 above)
Comment (topefolorunso) [reply airbyte-integrations/connectors/source-mailchimp/source_mailchimp/manifest.yaml:22]: It is correct as that is the implementation from the previous Python version

https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-mailchimp/source_mailchimp/source.py#L44-L47

This is how the access_token value is gotten

https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-mailchimp/source_mailchimp/spec.json#L75-L85

Comment (marcosmarxm) [reply airbyte-integrations/connectors/source-mailchimp/source_mailchimp/manifest.yaml:22]: You're looking in the wrong place
https://github.com/airbytehq/airbyte/blob/cae3ac4b715189ff3df5ef66766d62d57b1c14b8/airbyte-integrations/connectors/source-mailchimp/source_mailchimp/spec.json#L13-L42

File: airbyte-integrations/connectors/source-mailchimp/source_mailchimp/manifest.yaml [hunk 5]
```diff
@@ -0,0 +1,253 @@
...
+    request_headers: {}
+    path: '{{ parameters.path }}'
+    request_parameters:
+      sort_dir: ASC
+      sort_field: '{{ parameters.cursor_field }}'
+    authenticator:
+      class_name: source_mailchimp.components.MailchimpAuthenticator
+      basic_auth: "#/definitions/basic_authenticator"
+      oauth: "#/definitions/oauth_authenticator"
+    request_body_json: {}
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-mailchimp/source_mailchimp/manifest.yaml:36]: Sorry the missing return here @topefolorunso right now I'm talking to our engineer team about the declaration of nested custom components. Which is the case for this one.Any updates return to you here asap.

**Review by marcosmarxm — CHANGES_REQUESTED** [review]

**Comment by marcosmarxm** [comment]
Hello, and thank you so much for contributing to the project during the Hackathon. Due to the high complexity of the code migration, we still have a few contributions open. I'm sending this message to update you that this contribution will be added to the final points we're going to announce tomorrow. With that said, we ask for your help to continue the process and finish the review.

With the latest changes the connector is working but there are some sketchy happening during sync for `email_activities` I'll try to submit the final changes tomorrow @topefolorunso

**Comment by topefolorunso** [comment]
> With the latest changes the connector is working but there are some sketchy happening during sync for `email_activities` I'll try to submit the final changes tomorrow @topefolorunso

Thanks for the update.

**Review by marcosmarxm — APPROVED** [review]
Thanks @topefolorunso I needed to do some refactor to a few streams but now its working as the current version. I'll add a start date parameter the connector doesn't have today but its possible to use.

**Comment by topefolorunso** [comment]
> Thanks @topefolorunso I needed to do some refactor to a few streams but now its working as the current version. I'll add a start date parameter the connector doesn't have today but its possible to use.

Well done @marcosmarxm. What's the update on the requester? Does the nested authenticator work now?

**Comment by marcosmarxm** [comment]
@topefolorunso I opted to close the contribution for now. Until today I wasn't able to make the connector work with proper incremental syncs. I really thanks the contribution and effort you put into the code. See you next hackathon!

**Pull request closed without merging**